packages/opentelemetry-core/src/platform/node/hex-to-base64.ts

```diff
@@ -13,14 +13,30 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+function intValue(charCode: number): number {
+  // 0-9
+  if (charCode >= 48 && charCode <= 57) {
+    return charCode - 48;
+  }
+
+  // a-f
+  if (charCode >= 97 && charCode <= 102) {
+    return charCode - 87;
+  }
+
+  // A-F
+  return charCode - 55;
+}
+
 export function hexToBase64(hexStr: string): string {
-  const hexStrLen = hexStr.length;
-  let hexAsciiCharsStr = '';
-  for (let i = 0; i < hexStrLen; i += 2) {
-    const hexPair = hexStr.substring(i, i + 2);
-    const hexVal = parseInt(hexPair, 16);
-    hexAsciiCharsStr += String.fromCharCode(hexVal);
+  const buf = Buffer.alloc(hexStr.length / 2);
```

Comment:
The alloc is probably the most expensive part of this function at this point. Preallocating common lengths likely results in a significant speedup

```javascript
const buf8 = Buffer.alloc(8)
const buf16 = Buffer.alloc(16)

function hexToBase64(hexStr) {
    let buf;
    if (hexStr.length === 16) buf = buf8;
    else if (hexStr.length === 32) buf = buf16;
    else buf = Buffer.alloc(hexStr.length / 2);

    let offset = 0;

    for (let i = 0; i < hexStr.length; i += 2) {
        const hi = intValue(hexStr.charCodeAt(i));
        const lo = intValue(hexStr.charCodeAt(i + 1));
        buf.writeUInt8((hi << 4) | lo, offset++);
    }

    return buf.toString('base64');
}
```

Reply:
That's a good point, added in https://github.com/open-telemetry/opentelemetry-js/pull/3178/commits/95966fa0d47eee3c503e8943472ecf0dd2518ff4

Reply:
I used node-microbenchmark and the speedup is definitely real

```
Iterations: 375
  +-------------------------+--------------------+----------------------+-----------+
  | Name                    | nano seconds       | milli seconds        | Slower by |
  +-------------------------+--------------------+----------------------+-----------+
  | testHexToBase64Prealloc | 8349.088           | 0.008349088          | fastest   |
  | testHexToBase64         | 12648.157333333333 | 0.012648157333333333 | 1x        |
  +-------------------------+--------------------+----------------------+-----------+
```